### DISCUSSION TITLE
flags: rpc.batch.concurrency vs db.read.concurrency

### DISCUSSION THREAD

**Discussion by itgav** [erigontech/erigon#7336] [Q&A]
I'm using RPC requests to grab data from the Erigon database and am trying to optimize for speed. I've tested running the RPC daemon with the flags `--rpc.batch.concurrency` and `--db.read.concurrency` but haven't seen much impact on my RPC requests.

Any suggestions to optimize the speed of RPC requests?

**Comment by AskAlexSharov** [comment]
Too broad question, i will close it. What requests? What optimizing: latency or throughput? What machine/disk spec? Are you single consumer of rpc api or it’s shared api(can’t impact other users experience)? Why you need optimize something (current speed is not enough for what? Do you need 10% optimization or 1000%?)? Etc…

without this info, answer will be too broad: to optimize - need work on optimizations, see profiling data, find the bottleneck and fix it :-)

**Reply by itgav** [reply]
Ok thank you! I'm fairly noob so I don't always know what I don't know. I'll look into these and get back.

**Reply by AskAlexSharov** [reply]
@itgav didn't what to make you "feel as noob". have fun.
